pkg/config/config_test.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2020 The PipeCD Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestUnmarshalConfig(t *testing.T) {
+	testcases := []struct {
+		name     string
+		data     string
+		wantSpec interface{}
```

Comment:
Because testify cannot read private fields embedded in the struct, I determined to compare the `spec`s alone rather than the entire `Config{}`.